adding prompt tags and aliases (#2315)
wandb/docs: weave/guides/core-types/prompts-version.mdx

- Aliases are named pointers to a specific prompt version.
- Tags are labels applied to a specific prompt version.

To set aliases and tags for your prompts in the UI:
1. In the Weave project sidebar, click **Assets**. This opens the Assets page.
1. In the Assets page, click **Prompts**. Note that you can view assigned `Aliases` and `Tags` in the table.
1. In the Prompts table, click the link of the prompt you want to change.
1. In the prompt detail panel for your selected prompt, the title bar displays the name of the prompt and a specific version. Choose the version of the prompt you want to update.
1. In the panel toolbar, use the controls to add and remove assigned aliases and tags to this specific version.

![A selected prompt version in the prompt detail panel on the Prompts page, showing controls for Aliases and Tags.](/weave/guides/core-types/imgs/prompt-alias-tags.png)

### Modify tags and aliases in code

Tags and aliases can also be managed in code, such as publishing prompts with tags and aliases inline, pointing aliases at specific versions and resolving them to load prompts, and adding, removing, or listing tags and aliases on versions.

> The following code example will add a prompt `my-prompt` to your project with three versions, each having various tags and aliases. Update `'your-team-name/your-project-name'` accordingly for your project.

Reviewer: > project with three versions

I only see two versions (v0, and v1)

Reviewer: true. there are 3 publishes.

```
v0_ref = weave.publish(
    weave.StringPrompt("Answer the user's question: {question}"),
    name="my-prompt",
)
v1_ref = weave.publish(
    weave.StringPrompt("Answer the user's question helpfully and concisely: {question}"),
    name="my-prompt",
)

ref = weave.publish(
    weave.StringPrompt("Be brief: {question}"),
    name="my-prompt",
    tags=["reviewed"],
    aliases=["production"],
)
```

client.set_aliases(v1_ref, "production")

# Retrieve aliases for a version.
client.get_aliases(v1_ref)  # ["production", "latest"]

# Move an alias by setting it on a different version.
client.set_aliases(v0_ref, "production")  # Automatically detaches from v1.

# Resolve an alias to load the object.
prompt = weave.ref("my-prompt:production").get()

# List all aliases in the project.
client.list_aliases()

# Remove an alias.
client.remove_aliases(v0_ref, "production")

# --- Tags: labels on a specific version. ---

# Add tags to a version.
client.add_tags(v0_ref, ["reviewed", "passed-eval"])
client.add_tags(v1_ref, ["reviewed", "needs-improvement"])

# Get tags for a version.
client.get_tags(v0_ref)  # ["passed-eval", "reviewed"]

# List all distinct tags in the project.
client.list_tags()

# Remove tags from a version.
client.remove_tags(v1_ref, ["needs-improvement"])

# --- Combined: fetch both in a single call. ---

tags, aliases = client.get_tags_and_aliases(v0_ref)

# --- Publish with tags and aliases inline. ---

ref = weave.publish(
    weave.StringPrompt("Be brief: {question}"),
    name="my-prompt",
    tags=["reviewed"],
    aliases=["production"],
)

```

> ```plaintext Typescript lines

Reviewer: Should be `TypeScript`

5. (Optional) You can compare versions of prompts by clicking the checkboxes beside the listed prompts and then clicking the **Compare** button in the table toolbar. This allows you to see the diff between your prompts.

![The Compare prompts UI showing a diff between two different prompt versions.](/weave/guides/core-types/imgs/prompt-comparison.png)

> ## Prompt tags and aliases

Reviewer 2: ```suggestion
## Add prompt tags and aliases
```
Or "Add tags and aliases to prompts"?

You can organize prompt versions using aliases and tags. These labels help identify and reference specific versions.

- Aliases are named pointers to a specific prompt version.

> - Tags are labels applied to a specific prompt version.

Reviewer 2: Should we add context on when to use each of these? I feel like the definitions are a little circular without more explanation.

To set aliases and tags for your prompts in the UI:
1. In the Weave project sidebar, click **Assets**. This opens the Assets page.
1. In the Assets page, click **Prompts**. Note that you can view assigned `Aliases` and `Tags` in the table.
1. In the Prompts table, click the link of the prompt you want to change.
1. In the prompt detail panel for your selected prompt, the title bar displays the name of the prompt and a specific version. Choose the version of the prompt you want to update.
1. In the panel toolbar, use the controls to add and remove assigned aliases and tags to this specific version.

![A selected prompt version in the prompt detail panel on the Prompts page, showing controls for Aliases and Tags.](/weave/guides/core-types/imgs/prompt-alias-tags.png)

### Modify tags and aliases in code

> Tags and aliases can also be managed in code, such as publishing prompts with tags and aliases inline, pointing aliases at specific versions and resolving them to load prompts, and adding, removing, or listing tags and aliases on versions.

Reviewer 2: ```suggestion
Tags and aliases can also be managed in code, for tasks such as: 
- Publishing prompts with tags and aliases inline, 
- Pointing aliases at specific versions and resolving them to load prompts
- Adding, removing, or listing tags and aliases on versions.
```
nit. It just read a little heavy to me, otherwise. Feel free to ignore

> The following code example will add a prompt `my-prompt` to your project with three versions, each having various tags and aliases. Update `'your-team-name/your-project-name'` accordingly for your project.

Reviewer 2: ```suggestion
The following code example adds a prompt `my-prompt` to your project with three versions, each having various tags and aliases. Update `'your-team-name/your-project-name'` accordingly for your project.
```
Future tense
